src/nouveau/src/nouveau_httpd.erl

```diff
@@ -0,0 +1,276 @@
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+
+%% -*- erlang-indent-level: 4;indent-tabs-mode: nil -*-
+
+-module(nouveau_httpd).
+
+-include_lib("couch/include/couch_db.hrl").
+
+-export([
+    handle_analyze_req/1,
+    handle_search_req/3,
+    handle_info_req/3,
+    handle_cleanup_req/2
+]).
+
+-import(chttpd, [
+    send_method_not_allowed/2,
+    send_json/2, send_json/3,
+    send_error/2
+]).
+
+-define(RETRY_LIMIT, 20).
+-define(RETRY_SLEEP, 500).
+
+handle_analyze_req(#httpd{method = 'POST'} = Req) ->
```

Comment:
It would be good to have a basic integration test calling this API.

Reply:
sure